protocol/src/main/java/tw/waterball/survive/io/protocol/scanner/ModuleClassesScanner.java

```diff
@@ -0,0 +1,81 @@
+package tw.waterball.survive.io.protocol.scanner;
+
+import java.io.File;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.util.*;
+
+public final class ModuleClassesScanner {
+
+    private ModuleClassesScanner() {
+
+    }
+
+    public static Set<Class<?>> scanClassesByPackageName(String packageName) {
+        // 第一個class類的集合
+        Set<Class<?>> classes = new HashSet<>();
+        // 獲取包的名字並進行替換
+        String packageDirName = packageName.replace('.', '/');
+        try {
+            // 定義一個枚舉的集合併進行循環來處理這個目錄下的things
+            Enumeration<URL> dirs = Thread.currentThread().getContextClassLoader().getResources(packageDirName);
+            // 循環迭代下去
+            while (dirs.hasMoreElements()) {
+                // 獲取下一個元素
+                URL url = dirs.nextElement();
+                // 得到協議的名稱
+                String protocol = url.getProtocol();
+                // 如果是以文件的形式保存在服務器上
+                if ("file".equals(protocol)) {
+                    // 獲取包的物理路徑
+                    String filePath = URLDecoder.decode(url.getFile(), "UTF-8");
+                    // 以文件的方式掃描整個包下的文件並添加到集合中
+                    findClassesInPackageByFile(packageName, filePath, classes);
+                }
+            }
```

Comment:
Remove Chinese comments.
And add English comments to wherever you think it's necessary.